spyder/app/mainwindow.py

```diff
@@ -1221,6 +1221,15 @@ def pre_visible_setup(self):
                 except TypeError:
                     pass
 
+        # Register custom layouts
+        for plugin, plugin_instance in self._PLUGINS.items():
+            if hasattr(plugin_instance, 'CUSTOM_LAYOUTS'):
+                if isinstance(plugin_instance.CUSTOM_LAYOUTS, list):

```

Comment:
If `CUSTOM_LAYOUTS` is not a a list, perhaps we should print a warning saying that CUSTOM_LAYOUTS was detected for plugin `foo` but it can't be loaded because it's not a list.